src/features/metadata-based-view/MetadataBasedItemList.js

```diff
@@ -101,18 +111,50 @@ class MetadataBasedItemList extends React.Component<Props, State> {
         onItemClick(itemWithPreviewPermission);
     }
 
-    getGridCellData(columnIndex: number, rowIndex: number): GridCellData {
+    handleEditIconClick(columnIndex: number, rowIndex: number): void {
+        this.setState({
+            editedColumnIndex: columnIndex,
+            editedRowIndex: rowIndex,
+        });
+    }
+
+    handleCancelEdit = (): void => {
+        this.setState({
+            editedColumnIndex: -1,
+            editedRowIndex: -1,
+        });
+    };
+
+    handleSave = (): void => {
+        /* Implement me */
```

Comment:
Capturing user input, validating, and saving data will be covered by subsequent PR.